monkey/monkey_island/cc/server_utils/encryption/key_based_encryptor.py

```diff
@@ -21,25 +16,12 @@
 
 
 class KeyBasedEncryptor(IEncryptor):
```

Comment:
Let's take this opportunity to add docstrings to this class. Most importantly, the constructor needs to include details about what kind of key is acceptable. It might also be desirable to create a new type (class) called `EncryptionKey32Bit`. See [Parse, don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/).

Reply:
Here's an example of how to subclass `bytes`, https://github.com/ethereum/hexbytes/blob/master/hexbytes/main.py

Reply:
Another solution might be to implement something like [UserString](https://github.com/python/cpython/blob/07aeb7405ea42729b95ecae225f1d96a4aea5121/Lib/collections/__init__.py#L1336), but for bytes with a constraint that it contains exactly 4 bytes.